aten/src/ATen/EmptyTensor.cpp

```diff
@@ -3,6 +3,17 @@
 #include <ATen/detail/CUDAHooksInterface.h>
 #include <c10/core/CPUAllocator.h>
 
+// GCC has __builtin_mul_overflow from before it supported __has_builtin
+#ifdef __GNUC__
+#define HAS_BUILTIN_OVERFLOW() (1)
+#elif defined(__has_builtin)
+#define HAS_BUILTIN_OVERFLOW() (                    \
+      __has_builtin(__builtin_mul_overflow) &&      \
+      __has_builtin(__builtin_and_overflow))
+#else
+#define HAS_BUILTIN_OVERFLOW() (0)
+#endif
```

Comment:
Hmm, is't `#if !defined(_MSC_VER)` sufficient here?
```suggestion
#if !defined(_MSC_VER)
#define HAS_BUILTIN_OVERFLOW() (1)
#else
#define HAS_BUILTIN_OVERFLOW() (0)
#endif
```